resources/home/dnanexus/run_workflows/utils/dx_utils.py

```
@@ -0,0 +1,521 @@
+
+    # users specified in config to grant access to project
+    for user, access_level in users.items():
+        try:
+            project.invite(user, access_level, send_email=False)
+            prettier_print(f"\nGranted {access_level} privilege to {user}")
+        except Exception:
+            raise Exception(
+                Slack().send(
+                    f"Failed to grant {user} access to {project.name}\n{traceback.format_exc()}"
+                )
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)